llvm_mode/pass/AngoraPass.cc

```diff
@@ -524,9 +524,16 @@ void AngoraLLVMPass::resetIndirectCallContext(IRBuilder<> *IRB) {
 }
 
 void AngoraLLVMPass::processCall(Instruction *Inst) {
-  
-  CallInst *CI = dyn_cast<CallInst>(Inst);
-  Function* fp = CI->getCalledFunction();
+
+  Function* fp;
+  if (isa<CallInst>(Inst)) {
+    CallInst *CI = dyn_cast<CallInst>(Inst);
+    fp = CI->getCalledFunction();
+  }
+  else {
+    InvokeInst *II = dyn_cast<InvokeInst>(Inst);
```

Comment:
This should probably be a `cast` so the pass crashes with the correct error if `processCall` gets called from somewhere else in the future. Right now that would become a `nullptr` dereference.

Reply:
Stylistically speaking, maybe the following is better:
```
if (... dyn_cast ...)
else if (... dyn_cast ...)
else {
  llvm_unreachable(...)
}
```

Reply:
I changed the code to match your stylistic example. Please verify if this is what you meant.